### PR TITLE
Polymer 3 Conversion

### PR DESCRIPTION
**Pull request by apalaniuk**
Result of Polymer 3 conversion script. This leaves things in a non-functional state, I believe due to the PDF.js imports needing to be fixed up, but I believe that's to be done as a post-conversion step and saved as a patch.

### CODE REVIEW

**Comment by ChrisKraljevicD2L** [comment]
Fixups are indeed done as a post-conversion patch, but they can be included in the same PR as the actual results of the conversion script.

**Comment by ChrisKraljevicD2L** [comment]
I can't remember if `package-lock.json` needs to be updated, or removed from the repo pre-conversion. @dbatiste should know though.

File: d2l-pdf-viewer-toolbar-button.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			type: Boolean,
+			value: false
+		},
+		title: {
+			type: String
+		}
+	},
+
+	ready: function() {
+		this.$$('button').tabIndex = this.active ? 0 : -1;
```
Comment (ChrisKraljevicD2L) [inline d2l-pdf-viewer-toolbar-button.js]: Eeeeh I'm not sure if `this.$$` is still cool post polymer 3.

**Comment by dbatiste** [comment]
`package-lock.json` should be excluded in component repos.  We really just want the lock file in places where we want conscious decision to update dependencies, like BSI.

File: d2l-pdf-viewer-toolbar-button.js (hunk 1 above)
Comment (apalaniuk) [reply d2l-pdf-viewer-toolbar-button.js]: Why not? The converter/modulizer adds `@polymer/polymer/polymer-legacy.js`, which pulls in the `LegacyElementMixin`

Comment (ChrisKraljevicD2L) [reply d2l-pdf-viewer-toolbar-button.js]: Ah, alright, my mistake.

**Comment by apalaniuk** [comment]
Closing in favour of an upcoming PR in the newly-created https://github.com/Brightspace/d2l-pdf-viewer

**Pull request closed without merging**